design-proposals/bound-token-for-vnc.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+
+Kubernetes API server can create 
+[tokens](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens)
+bound to a service account. They are mainly used in services running inside the cluster to authenticate
+between each other. 
+
+This document describes how to create a token that can be used to authenticate with the API server
+to only access VNC endpoint of a singe VM.
+
+## Motivation
+
+An administrator may want to give VNC access to a client application without giving it permissions to access
```

Comment:
So instead of granting a specific user access to the `/vnc` subresource of a given `VirtualMachineInstance` we instead provide the user access to the `/vnc-token` subresource of the same `VirtualMachineInstance` to generate a token they can then use to access the former for a specific amount of time?

I assume a RBAC role per mapping isn't an option here because users couldn't use it to unlock access to their specific `VirtualMachineInstance` for another user?

Reply:
I'm not sure I understand your question. A user can have access to the `/vnc` subresource. The token will be used to give access to `/vnc` for non-registered users. When an API request with the token is made, it will be authenticated as a specific ServiceAccount, which only has permissions to access `/vnc`.

Reply:
Apologies I mixed client with an actual registered user of the cluster, you do go on to highlight that again below.